document Re-LAION-1300K (#32)
finegrain-ai/LBM: contrib/eraser/README.md

**Goal**: reproduce the object-removal result from the LBM paper (Section 4.1) using SD1.5.
Non-official implementation.

## Install

> One option is to install [uv](https://docs.astral.sh/uv/getting-started/installation/) and then:

Reviewer: ```suggestion
One option is to install [uv](https://docs.astral.sh/uv/getting-started/installation/) and then, from the root folder:
```

Reviewer: just to be clear

uv run --script contrib/eraser/datasets/extract_val_lite.py
```

## Re-LAION Dataset

> We extract 1.3M images from [Re-LAION-Caption19M](https://huggingface.co/datasets/supermodelresearch/Re-LAION-Caption19M)

Reviewer: ```suggestion
We extract 1.3M images from [Re-LAION-Caption19M](https://huggingface.co/datasets/supermodelresearch/Re-LAION-Caption19M) for inpainting (see paper 4.1: "in-the-wild images where we randomly masked an area of the image")
```

### Requirements

> 1/ Install duckdb CLI

Reviewer: ```suggestion
1/ Install [duckdb](https://duckdb.org/) CLI
```

> ```

Reviewer: ```suggestion
```bash
```

curl https://install.duckdb.org | sh
```

> 2/ Install img2dataset

Reviewer: ```suggestion
2/ Install [img2dataset](https://github.com/rom1504/img2dataset)
```

> ```

Reviewer: ```suggestion
```bash
```

> uv pip install img2dataset

Reviewer: Given that `uv` is used throughout this README, I think we should add a section right after the intro, something like:

## Install

One option is to install [uv](https://docs.astral.sh/uv/getting-started/installation/) and then:

```bash
uv venv -p 3.10
uv pip sync requirements.txt
uv pip install -e .
```

--

That way it's clear this is what we are using. Also (nit) you could change the train command: `uv run python contrib/...` for consistency with the other commands.

> uv pip install img2dataset

Reviewer: Why don't you just install it as a tool?

```bash
uv tool install -p 3.10 img2dataset
```

```

3/ Set a knot resolver by following [img2dataset instructions](https://github.com/rom1504/img2dataset?tab=readme-ov-file#setting-up-a-knot-resolver)

### Download

First download the files

> ```

Reviewer: ```suggestion
```bash
```

git lfs install
git clone git@hf.co:datasets/supermodelresearch/Re-LAION-Caption19M data/supermodelresearch/Re-LAION-Caption19M
```
Then extract the images with `aestetic > 0.5` and `pwatermark < 0.2` (takes ~50s)

> ```

Reviewer: ```suggestion
```bash
```

duckdb < contrib/eraser/datasets/re_laion_caption19m_filter.sql
```

> An extract of `Re-LAION-Caption19M` is then saved in `Re-LAION-1300K-parquet`

Reviewer: ```suggestion
An extract of `Re-LAION-Caption19M` is then saved in `data/Re-LAION-1300K-parquet`
```

> It takes ~4h (on a `c6i.32xlarge`) and it requires ~1.1TB of free space disk

Reviewer: ```suggestion
It takes ~4h (on a `c6i.32xlarge`) and it requires ~1.1TB of free space disk to convert into WebDataset format
```

> ```

Reviewer: ```suggestion
```bash
```

export DATA_FOLDER='data/Re-LAION-1300K-parquet/'
export OUTPUT_FOLDER=data/Re-LAION-1300K
export SAVE_ADDITIONAL_COLUMNS='["aesthetic_score","llava_next_caption","luminance_score","ocr_score","llava_next_caption_shuffled","similarity","punsafe","pwatermark"]'
export PROCESSES_COUNT=128
export THREAD_COUNT=256

NO_ALBUMENTATIONS_UPDATE=1 uv run img2dataset \
  --url_list $DATA_FOLDER \
  --output_folder $OUTPUT_FOLDER \
  --input_format parquet \
  --output_format webdataset \
  --processes_count $PROCESSES_COUNT\
  --thread_count $THREAD_COUNT\
  --min_image_size 256 \
  --number_sample_per_shard 1024 \
  --resize_mode no \
  --max_shard_retry 5 \
  --max_image_area 16777216 \
  --save_additional_columns $SAVE_ADDITIONAL_COLUMNS
```

Notes : 

> * We purposely do not resize the images here so the dataset is compatible with multiple size

Reviewer: ```suggestion
* We purposely do not resize the images here so the dataset is compatible with multiple sizes
```

* Setting `--resize_mode no` is mandatory, by default it's `--resize_mode border` with `--image_size 256`
* Setting `--max_image_area 16777216 = 4096x4096` is made to avoid errors like `PIL.Image.DecompressionBombError`

> * Currently we only set `--min_image_size 256` as a sanity check, but the dataset is supposed to contains only 1024 images already

Reviewer: ```suggestion
* Currently we only set `--min_image_size 256` as a sanity check, but the dataset is supposed to contain only 1024 images already
```

> * If the script is hanging, just kill it and re-run it, it's resuming smoothly

Reviewer: ```suggestion
* If the script is hanging, just kill it and re-run it, it'll resume smoothly
```

```bash
curl https://install.duckdb.org | sh
```

2/ Install [img2dataset](https://github.com/rom1504/img2dataset)
```bash
uv tool install -p 3.10 img2dataset
```

3/ Set a knot resolver by following [img2dataset instructions](https://github.com/rom1504/img2dataset?tab=readme-ov-file#setting-up-a-knot-resolver)

### Download

First download the files
```bash
git lfs install
git clone git@hf.co:datasets/supermodelresearch/Re-LAION-Caption19M data/supermodelresearch/Re-LAION-Caption19M
```
Then extract the images with `aestetic > 0.5` and `pwatermark < 0.2` (takes ~50s)
```bash
duckdb < contrib/eraser/datasets/re_laion_caption19m_filter.sql
```
An extract of `Re-LAION-Caption19M` is then saved in `data/Re-LAION-1300K-parquet`

It takes ~4h (on a `c6i.32xlarge`) and it requires ~1.1TB of free space disk to convert into WebDataset format
```bash
export DATA_FOLDER='data/Re-LAION-1300K-parquet/'
export OUTPUT_FOLDER=data/Re-LAION-1300K
export SAVE_ADDITIONAL_COLUMNS='["aesthetic_score","llava_next_caption","luminance_score","ocr_score","llava_next_caption_shuffled","similarity","punsafe","pwatermark"]'
export PROCESSES_COUNT=128
export THREAD_COUNT=256

> NO_ALBUMENTATIONS_UPDATE=1 uv run img2dataset \

Reviewer: ```suggestion
NO_ALBUMENTATIONS_UPDATE=1 img2dataset \
```

Reviewer: should just work since uv tool make it available in your path.
